src/deploy/osp_deployer/director.py

```diff
@@ -1256,13 +1256,18 @@ def setup_nic_configuration(self):
             cmds.append('sed -i -r \'s/(^\s*' + setting_name +    # noqa: W605
                         ':\s*).*/\\1' + setting_value + '/\' ' + remote_file)
 
+        # if smart NICs are used then use smart configuration for contoller and compute nodes
+        if self.settings.enable_smart_nic == True and self.get_sriov_interfaces() == 2:
```

Comment:
We really shouldn't be modifying the files that the nic_environment.yaml points to on the fly.  Instead, we should just add a new sriov_6_port_smart NIC configuration